### PR TITLE
Introduce support for named parameters in query using @name syntax.

### LINKED ISSUE #54: does not handle sql comments with question mark
I had a large query that had a question mark in the SQL comments (after --).  This resulted in an error "sql:  expected 1 arguments, got 0".  It took me a while to track this down, but the question mark is used by this driver's NumInput() function to indicate the presence of an argument.  As alexbrainman/odbc doesn't have this issue I suspect there is another way to evaluate this property. For now I have removed the question mark and the query seems to be working correctly.

### LINKED ISSUE #60: Interpolation with Named Paramters
I am trying to run a query using named parameters like this:

```sql
select
*
from
some_table
where
client like @CLIENT_NAME
```

The `@` symbol appears to be the correct syntax based on the docs [here](https://www.vertica.com/docs/9.2.x/HTML/Content/Authoring/ConnectingToVertica/ClientADO/UsingParameters.htm) but when I run this with this adapter like this:
```go
queryTerm := "test"
rows, err := stmt.QueryContext(ctx, sql.Named("CLIENT_NAME", queryTerm))
```

I get `Error: [42703] Column "CLIENT_NAME" does not exist`

### PR DESCRIPTION
**Pull request by mlh758**
Use a lexer to handle parameters
Fixes #54 by handling comments and strings.
Closes #60 by adding support for named parameters

### CODE REVIEW

**Comment by mlh758** [comment]
Hm, something isn't quite right. I ran this with a much larger query and it's complaining about not enough arguments.

**Comment by mlh758** [comment]
Looks like the error is coming from [here](https://github.com/golang/go/blob/master/src/database/sql/convert.go#L201) I''ll look into how to make the sql library happy.

**Comment by mlh758** [comment]
With this PR `@param`, `@PARAM`, and `@Param` would be 3 different variable names while SQL is usually not case sensitive. Should I normalize them all to upper or lower case before passing them out of the lexer? Or do you think having `@Param` appear as `@PARAM` would be more confusing?

**Comment by huebnerr** [comment]
@mlh758, first off, some beautiful and clean code there. IMHO, internally normalizing them into the same thing (upper or lower case, your call) would be nicer. I don't think those are generally case sensitive. I've got a tight schedule today, so probably won't be able to review it until tomorrow.

Quick question, is there a test case for using named parameters with prepared statements? I'm pretty sure the internal handling by the sql library would be exactly the same.

Also, please make sure to add your name to a "Special Thanks" section in the doc. There are several people that have done yeoman's work in adding new features and identifying bugs that are memorialized.

**Comment by mlh758** [comment]
I'll normalize them all to upper case, and fix these tests I just broke.

Do you mean have I added a test to the driver_test file that actually hits the database? If so, I have not but I'm sure I could. I've been doing my end-to-end testing against a Vertica database at work.

How would you feel about adding a build tag of `// +build integration` to `driver_test.go` and renaming it `driver_int_test.go`? That way it would only hit a database when you ran it as `go test --tags=integration`. I wouldn't do that in this PR since I have enough changes as it is and it would require a change to your CI. The advantage would be that contributors could run `go test ./...` as normal and run all the unit tests in this package without needing to set up a vertica database for development.

**Comment by mlh758** [comment]
Added some integration tests for the named parameters to `driver_test.go` I apologize if this leads to some commit spam, I can't run them locally.

@huebnerr where is the special thanks section?

File: parse/queryLex.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	input        string
+	pos          int
+	start        int
+	width        int
+	onNamed      OnNamedParam
+	onPositional SubstitutePosParam
+	output       strings.Builder
+}
+
+// LexOption is a function that sets a prefernce on the lexer
```
[the remark this answers is not in the document]
Comment (mlh758) [reply parse/queryLex.go]: Fixed the typo

**Comment by mlh758** [comment]
Also, thank you for the compliment about the code quality but I can't take all the credit. I based the lexer's structure off a talk by [Rob Pike](https://www.youtube.com/watch?v=HxaD_trXwRE).

**Comment by huebnerr** [comment]
Looks great. Thanks for the very valuable contribution!

**Pull request merged**